### DISCUSSION TITLE
Pause Until Application Launches/Activates? Issue With Moom

### DISCUSSION THREAD

**Discussion by alexfazio** [ttscoff/bunch-docs#231] [Q&A]
Correct me if I am wrong. I could not find in the official docs any syntax to delay the next command until an application launches or activates. Does this kind of command exist?

In my use case, some apps are slow to launch, which creates a conflict with **Moom.app**. Moom tries to arrange windows straight away, when commanded so by Bunch, and doesn't find them available.

I am aware of the `- (pause 10)` command. Yet it's a pretty inelegant and inefficient solution, which can still create problems if the launch time varies for whatever reason.

**Comment by ttscoff** [comment]
It's not entirely possible to determine when all said have launched, and even more difficult to know if they've displayed a window (which is what Moom needs) but a combination of a waiting snippet and a short delay works well.

https://bunchapp.co/docs/bunch-files/snippets/#waitingsnippet

-Brett
________________________________
From: alexfazio ***@***.***>
Sent: Sunday, July 17, 2022 3:27:56 AM
To: ttscoff/bunch ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [ttscoff/bunch] Pause Until Application Launches/Activates? Issue With Moom (Discussion #231)


Correct me if I am wrong. I could not find in the official docs any syntax to delay the next command until an application launches or activates. Does this kind of command exist?

In my use case, some apps are slow to launch, which creates a conflict with Moom.app. Moom tries to arrange windows straight away, when commanded so by Bunch, and doesn't find them available.

I am aware of the - (pause 10) command. Yet it's a pretty inelegant and inefficient solution, which can still create problems if the launch time varies for whatever reason.
